src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -101,19 +101,31 @@ public class SqlMutationEngine : IMutationEngine
             // If authorization fails, an exception will be thrown and request execution halts.
             AuthorizeMutationFields(context, parameters, entityName, mutationOperation);
 
+            string roleName = GetHttpContext().Request.Headers[AuthorizationResolver.CLIENT_ROLE_HEADER];
```

Comment:
see `AuthorizeMutationFields()` later in this file. I suggest breaking out the code that resolves the role into a separate private helper which can be reused here. Essentially, resolving the role should occur by pulling the client role header out of the IMiddlewareContext and not the HTTP context.

Reply:
I've added a helper method and updated the code here and within AuthorizeMutationFields() to make use of the helper method for role determination